### DISCUSSION TITLE
Another problem with inconsistency checking

### DISCUSSION THREAD

**Discussion by tomkolp** [WeblateOrg/weblate#5725] [General] [answered]
After adding new files and translating automatically, the consistency check does not work as expected.

1 Adds several components that share common segments (same context and source)
2 I run automatic translation manually

Common strings are marked as an inconsistency. After the automatic translation is finished, there is no more inconsistency, but the check is still there. I have to manually run checking for each file separately. Checking the entire project is not necessary because it takes several days. Usually adds more than 100 files at once.

I think calling checks for a specific component list would solve the problem.

![image](https://user-images.githubusercontent.com/39633242/112117406-a054f680-8bbb-11eb-844a-0888f0f248c5.png)

![image](https://user-images.githubusercontent.com/39633242/112117590-cc707780-8bbb-11eb-8df3-7ea4d966d869.png)

![image](https://user-images.githubusercontent.com/39633242/112117755-f75acb80-8bbb-11eb-9fb2-be248c58490d.png)

**Comment by nijel** [comment, marked as answer]
I've added tests to cover this in 241901e5c70a3fbb4339e5b971101a6c3aa5dd32, and it seems to work correctly there.

**Reply by tomkolp** [reply]
The problem is from entering "batch". When I run the check manually, it fixes itself.

**Reply by nijel** [reply]
The code I've added does test the batch checks update. Is translation propagation enabled on all your components?

**Reply by tomkolp** [reply]
Propagation is enabled throughout the project. This only happens after manually triggered automatic translation. There is no problem with normal translation.
I got the names of all the components from the list via api and created a script that manually runs the checks:

```
#! / bin / sh
docker-compose exec --user weblate weblate weblate updatechecks eso-crack / reapers-march-ezzags-bandits
docker-compose exec --user weblate weblate weblate updatechecks eso-crack / reapers-march-how-few-remain
.....
.....
```

This fixed all the checks in the entire component list.

**Reply by nijel** [reply]
The check’s update is supposed to happen in the background, so it might take some time. Or it might crash and that would appear in the logs only (unless you have configured Sentry).

**Reply by tomkolp** [reply]
I have sentry configured, I did not notice any errors a few hours after automatic translation.
The next time I add new files I will wait longer and see if the checking starts.

**Reply by tomkolp** [reply]
Today five hours ago, I used automatic translation on 60 files.  The files contain some strings with the same context and source as files already present in the project.  After 5 hours, checking was not started.  Sentry did not record any problems at this time.  I have the latest version bleeding from today.
Edit:
After 14 hours, the check started. Is this time so long because the project is huge? Ultimately, the project will be three times larger.
